src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/TruthSensitivityTranche.java

```diff
@@ -0,0 +1,253 @@
+
+        Collections.sort(data, VariantDatum.VariantDatumLODComparator);
+        metric.calculateRunningMetric(data);
+
+        List<TruthSensitivityTranche> tranches = new ArrayList<>();
+        for ( double trancheThreshold : trancheThresholds ) {
+            TruthSensitivityTranche t = findTranche(data, metric, trancheThreshold, model);
+
+            if ( t == null ) {
+                if (tranches.isEmpty()) {
+                    throw new UserException(String.format("Couldn't find any tranche containing variants with a %s > %.2f. Are you sure the truth files contain unfiltered variants which overlap the input data?", metric.getName(), metric.getThreshold(trancheThreshold)));
+                }
+                break;
+            }
+
+            tranches.add(t);
+        }
+
+        tranches.sort(TruthSensitivityTranche.TRUTH_SENSITIVITY_ORDER);
+        return tranches;
+    }
+
+    private static TruthSensitivityTranche findTranche(final List<VariantDatum> data, final TruthSensitivityMetric metric, final double trancheThreshold, final VariantRecalibratorArgumentCollection.Mode model ) {
+        logger.debug(String.format("  TruthSensitivityTranche threshold %.2f => selection metric threshold %.3f", trancheThreshold, metric.getThreshold(trancheThreshold)));
+
+        double metricThreshold = metric.getThreshold(trancheThreshold);
+        int n = data.size();
+        for ( int i = 0; i < n; i++ ) {
+            if ( metric.getRunningMetric(i) >= metricThreshold ) {
+                // we've found the largest group of variants with sensitivity >= our target truth sensitivity
+                TruthSensitivityTranche t = trancheOfVariants(data, i, trancheThreshold, model);
+                logger.debug(String.format("  Found tranche for %.3f: %.3f threshold starting with variant %d; running score is %.3f ",
+                        trancheThreshold, metricThreshold, i, metric.getRunningMetric(i)));
+                logger.debug(String.format("  TruthSensitivityTranche is %s", t));
+                return t;
+            }
+        }
+
+        return null;
+    }
+
+    protected static TruthSensitivityTranche trancheOfVariants(final List<VariantDatum> data, final int minI, final double ts, final VariantRecalibratorArgumentCollection.Mode model ) {
+        int numKnown = 0, numNovel = 0, knownTi = 0, knownTv = 0, novelTi = 0, novelTv = 0;
+
+        final double minLod = data.get(minI).lod;
+        for ( final VariantDatum datum : data ) {
+            if ( datum.lod >= minLod ) {
+                if ( datum.isKnown ) {
+                    numKnown++;
+                    if( datum.isSNP ) {
+                        if ( datum.isTransition ) { knownTi++; } else { knownTv++; }
+                    }
+                } else {
+                    numNovel++;
+                    if( datum.isSNP ) {
+                        if ( datum.isTransition ) { novelTi++; } else { novelTv++; }
+                    }
```

Comment:
It looks like this code is all copied below for VQSLODTranche.  Could you please push it up into the Tranche class instead?